eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/actions/TicketButtonAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.actions;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.NonNull;
+
+import com.schedjoules.client.eventsdiscovery.Event;
+import com.schedjoules.eventdiscovery.R;
+import com.schedjoules.eventdiscovery.framework.utils.AbstractCachingOptional;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+import org.dmfs.httpessentials.types.Link;
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * {@link Action} for the Ticket button.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TicketButtonAction extends AbstractCachingOptional<Action>
+{
+
+    public TicketButtonAction(final Optional<Link> ticketLink, final Event event)
+    {
+        super(new Factory<Optional<Action>>()
+        {
+            @Override
+            public Optional<Action> create()
+            {
+                if (!ticketLink.isPresent())
```

Comment:
Created `Mapped` Optional for that and kept `TicketButtonAction` which now hides `BookTicketLink` as well.